src/search.js

```diff
@@ -1,31 +1,31 @@
-import {assoc, concat, filter, has, isEmpty, map, pipe, prop, sortBy} from "ramda"
+import {ascend, assoc, concat, descend, has, isEmpty, map, prop, sortBy, sortWith} from "ramda"
 import * as diacritics from 'diacritics'
 
 
 export function findParametersAndVariables(parameters, variables, query) {
-  function matchesQuery(parameterOrVariable) {
-    if (isEmpty(normalizedQuery)) {
-      return true
-    }
-    return parameterOrVariable.name.includes(normalizedQuery) ||
-      parameterOrVariable.searchIndex.includes(normalizedQuery)
+  function weightByPositionIn(string, substring) {
+    const index = string.indexOf(substring)
+    return index === -1 ? Number.MAX_SAFE_INTEGER : index
   }
   const normalizedQuery = diacritics.remove(query.toLowerCase())
-  return pipe(
-    filter(matchesQuery),
-    sortBy(prop('name')),
-  )(concat(
-    map(assoc('type', 'parameter'), parameters),
-    map(assoc('type', 'variable'), variables),
-  ))
+  const parametersAndVariables = concat(parameters, variables)
+  if (isEmpty(normalizedQuery)) {
+    return sortBy(prop('name'), parametersAndVariables)
+  } else {
```

Comment:
No, the ternary and the early return are different implementations. Ternaries for long values are unreadable.

What I'm saying is you can write the discussed piece of code as follows:

```js
if (isEmpty(normalizedQuery))
    return sortBy(prop('name'), parametersAndVariables)

// do something that's in the `else` block: it will only be reached when the `if` block was not reached
```